nagios.go

```diff
@@ -0,0 +1,113 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os/exec"
+	"regexp"
+	"strings"
+)
+
+// NagiosData is a task factory for tasks that fetch a JSON representation of
+// the provided project. The task uses outFor and errOutFor to get io.Writers
+// to write, respectively, the JSON output and any eventual error message.
+func NagiosData(project string, outFor, errOutFor projectResourceWriterCloserFactory) Task {
+	return nagiosData(func(project, pod string) *exec.Cmd {
+		return exec.Command("oc", "exec", pod, "cat", "/var/log/nagios/status.dat")
+	}, project, outFor, errOutFor)
+}
+
+// A getNagiosDataCmdFactory generates commands to get the raw nagios status data
+// in the provided project from the provided pod
+type getNagiosDataCmdFactory func(project, pod string) *exec.Cmd
+
+// nagiosData will dump a JSON representation of the current nagios status into
+// the io.Writer returned from outFor and log any errors it encounters doing so
+// into the io.Writer returned from errOutFor.
+func nagiosData(cmdFactory getNagiosDataCmdFactory, project string, outFor, errOutFor projectResourceWriterCloserFactory) Task {
+	return func() error {
+		stdout, stdoutCloser, err := outFor(project, "nagios")
+		if err != nil {
+			return err
+		}
+		defer stdoutCloser.Close()
+
+		stderr, stderrCloser, err := errOutFor(project, "nagios")
+		if err != nil {
+			return err
+		}
+		defer stderrCloser.Close()
+
+		var stdoutBuff bytes.Buffer
+
+		nagiosPod, err := getProjectPodNamesBySubstr(project, "nagios")
+		if err != nil {
+			fmt.Fprint(stderr, err)
+			return err
+		}
+
+		if len(nagiosPod) == 0 {
```

Comment:
@vchepeli as far as I know, go-lang does not have a method like `isEmpty` for arrays / slices and `len(slice)` is the standard way to check for this case. @rhcarvalho @maleck13 Could I get your opinion on this?